diff -up grfcodec/getopt.c.orig grfcodec/getopt.c
--- grfcodec/getopt.c.orig	2006-09-06 00:22:55.000000000 +0200
+++ grfcodec/getopt.c	2009-08-24 17:46:44.000000000 +0200
@@ -157,7 +157,7 @@ static enum
 
 char *getenv ();
 
-static char *my_index (const char *string, int chr)
+static const char *my_index (const char *string, int chr)
 {
   while (*string)
     {
@@ -520,7 +520,7 @@ int _getopt_internal (int argc, char *co
 
   {
     char c = *nextchar++;
-    char *temp = my_index (optstring, c);
+    const char *temp = my_index (optstring, c);
 
     /* Increment `optind' when we start to process its last character.  */
     if (*nextchar == '\0')
diff -up grfcodec/grfdiff.cc.orig grfcodec/grfdiff.cc
--- grfcodec/grfdiff.cc.orig	2009-07-25 20:58:43.000000000 +0200
+++ grfcodec/grfdiff.cc	2009-08-24 17:47:56.000000000 +0200
@@ -253,7 +253,7 @@ void grfstore::getorgdata(U16 *size, U8 
 	fseek(grf, fpos_cur, SEEK_SET);
 }
 
-char *basename(const char *filename)
+char *grfbasename(const char *filename)
 {
 	static char bname[32];
 	fnsplit(filename, NULL, NULL, bname, NULL);
@@ -389,7 +389,7 @@ int mkdiff()
 	}
 
 	if (grd) {
-		char *noext = basename(grffile[1]);
+		char *noext = grfbasename(grffile[1]);
 		char *p;
 
 		for (p = noext; *p; p++) { *p = tolower(*p); }
diff -up grfcodec/readinfo.cc.orig grfcodec/readinfo.cc
--- grfcodec/readinfo.cc.orig	2009-08-03 21:49:30.000000000 +0200
+++ grfcodec/readinfo.cc	2009-08-24 17:47:25.000000000 +0200
@@ -39,6 +39,7 @@ Version 7: Add backslash escapes
 #include<string>
 #include<sstream>
 #include<iomanip>
+#include<cstdio>
 
 using namespace std;
 
