Index: pseudo.cpp
===================================================================
--- pseudo.cpp	(revision 2184)
+++ pseudo.cpp	(working copy)
@@ -23,6 +23,7 @@
 #include<iostream>
 #include<iomanip>
 #include<cstdarg>
+#include<fstream>
 
 /* If your compiler errors on the following line, boost is not
  * properly installed.
Index: getopt.c
===================================================================
--- getopt.c	(revision 2184)
+++ getopt.c	(working copy)
@@ -498,7 +498,7 @@
 
   {
     char c = *nextchar++;
-    char *temp = my_index (optstring, c);
+    const char *temp = my_index (optstring, c);
 
     /* Increment `optind' when we start to process its last character.  */
     if (*nextchar == '\0')
